### ISSUE TITLE
Connect with Heroku postgres

### ISSUE THREAD

**Issue by noloop** [izniburak/pdox#61]
I am working with Heroku and I use pdox in my app. I'm having trouble connecting to Heroku's postgres database using Pdox, I believe the solution would be to pass the connection Heroku gives me so that Pdox can work, I see how Heroku works:

```php
<?php
# This function reads your DATABASE_URL configuration automatically set by Heroku
# the return value is a string that will work with pg_connect
function pg_connection_string() {
  return "dbname=abcdefg host=****.amazonaws.com port=5432 user=**** password=**** sslmode=require";
}
 
# Establish db connection
$db = pg_connect(pg_connection_string());
if (!$db) {
    echo "Database connection error."
    exit;
}
 
$result = pg_query($db, "SELECT statement goes here");
```

I needed to pass `$db` to Pdox instead of a `$config`, which is the best way I could do this:

```php
$dbPdox = new \Buki\Pdox( $db);
```

**Comment by izniburak** [comment]
Hi @noloop ,
I see your problem. Maybe, we can make a such feature. But you should still use PDO for this. Because, PDOx library is using PDO object to connect Database and its all operations.

If you can use PDO object to connect your database, we can make some improvements about that.

It is make sense for you?

Thanks.

**Comment by noloop** [comment]
Hi @izniburak ,
I can use the PDO yes, see:

```php
$db = (function(){
    $parts = (parse_url(getenv('DATABASE_URL') ?: 'postgresql://username:password@localhost:5432/my-database-name'));
    extract($parts);
    $path = ltrim($path, "/");
    return new PDO("pgsql:host={$host};port={$port};dbname={$path}", $user, $pass);
})();

```
How would PDOx use this PDO object $db instead of $config?

**Comment by noloop** [comment]
I try connect so:

```php
<?php
$config = [
	'host' =>  'myaddress.compute-1.amazonaws.com:5432',
	'driver' =>  'pgsql',
	'database' => 'mybdname',
	'username' => 'myusername',
	'password' => 'mypass',
	'charset' => 'utf8',
	'collation' => 'utf8_general_ci',
	'prefix' => ''
];
$db = new \Buki\Pdox($config);
```
I receiving:

> Cannot the connect to Database with PDO. SQLSTATE[08006] [7] could not translate host name "myaddress.compute-1.amazonaws.com:5432" to address: Name or service not know

**Comment by noloop** [comment]
I try again so, without port:

```php
<?php
$config = [
	'host' =>  'myaddress.compute-1.amazonaws.com',
	'driver' =>  'pgsql',
	'database' => 'mybdname',
	'username' => 'myusername',
	'password' => 'mypass',
	'charset' => 'utf8',
	'collation' => 'utf8_general_ci',
	'prefix' => ''
];
$db = new \Buki\Pdox($config);
```

And worked now. But there is one bad news for me, postgres ins't case sensitive, all my model code write in camelCase. I have a big work now.

**Comment by izniburak** [comment]
Hi @noloop 
So, has the problem been solved?

**Comment by noloop** [comment]
@izniburak yes, but there is a new problem that has appeared, "AS" is not working in "psql" mode, this should like this? or something is wrong?

**Comment by izniburak** [comment]
@noloop you mean i guess "as" is alias keyword, right? If this keyword is a problem for Postgre, i can make some improvements about that. What do we have to do? Maybe we can remove "as" keyword in the query which will be generated when we use Postgre.

**Comment by noloop** [comment]
@izniburak Example in a postgres database with a table called "category" with two fields "idcategory" and "namecategory" without camelCase because postgres doesn't case sensitive when I do try:

```php
$rows = $this->conn->select(['idCategory', 'nameCategory'])
                       ->table('Category')
                       ->orderBy('idCategory')
                       ->getAll();
```

It works even though I select with camelCase, but the $rows array gives me objects with properties whihout camelCase, like this:

```php
foreach($rows as $key => $value) {
  $categories[$value->namecategory] = $value->idcategory ;
}
```

And if I try to change the names to:

```php
$rows = $this->conn->select(['idCategory AS idCategory', 'nameCategory AS nameCategory'])
                       ->table('Category')
                       ->orderBy('idCategory')
                       ->getAll();
```

or

```php
$rows = $this->conn->select(['idcategory AS idCategory', 'namecategory AS nameCategory'])
                       ->table('Category')
                       ->orderBy('idCategory')
                       ->getAll();
```

does not work yet the generated object gives me names without camelCase.

**Comment by izniburak** [comment]
Actually, PDOx returns datas what you specified directly. I mean, if you specified a column as camelCase, it will return as camelCase. I'm try to understand the problem. If I find any issue regarding to this, I will try to fix it.

Thanks.

**Comment by noloop** [comment]
@izniburak It's not returning camelCase, it's returning without camelcase, because the postgres database doesn't support camelCase, but I'm calling with camelCase and it still works, which shouldn't, so I try to rename it and it doesn't work, still returning me without camelCase . This only happens with pgsql, with mysql everything works as expected.

**Comment by izniburak** [comment]
@noloop do you have any advice for this? If postgre don't have camelCase support, why do you try to use camelCase? I guess i could not understand the problem completely.

**Comment by noloop** [comment]
@izniburak In postegre I have a table named "category" with a field "namecategory", the problem is that selecting "nameCategory" in the table "Category" works and returns me an array with objects with the property "namecategory", it should not work since I do not have any table in the "Category" post that has a "nameCategory" property.

**Comment by noloop** [comment]
I'm in my database:

```php
CREATE TABLE "Category" (
    "idCategory" SERIAL,
    "nameCategory" VARCHAR(255) NOT NULL,
    CONSTRAINT "PK_Category"
        PRIMARY KEY ("idCategory"));
```

When I use PDOX for get all categories:

```php
 $rows = $this->conn->select(['idCategory', 'nameCategory'])
                       ->table('Category')
                       ->orderBy('idCategory')
                       ->getAll();
    var_dump($rows);
```

I received:

```php
array(4) {
  [0]=>
  object(stdClass)#23 (2) {
    ["idcategory"]=>
    int(1)
    ["namecategory"]=>
    string(7) "Romance"
  }
  [1]=>
  object(stdClass)#24 (2) {
    ["idcategory"]=>
    int(2)
    ["namecategory"]=>
    string(7) "Poesias"
  }
  [2]=>
  object(stdClass)#25 (2) {
    ["idcategory"]=>
    int(3)
    ["namecategory"]=>
    string(6) "Contos"
  }
  [3]=>
  object(stdClass)#26 (2) {
    ["idcategory"]=>
    int(4)
    ["namecategory"]=>
    string(5) "Artes"
  }
}
```

See the object fields in lower case and not camelCase.

**Comment by izniburak** [comment]
I think, the problem is not related to PDOx. I guess PDO is converting all column to lowercase from camelCase. For now I don't know what can we do for this. I will research.
Thanks.

**Comment by noloop** [comment]
@izniburak I think the problem is that PDOX is not using quotes to fetch columns from the database, so PDO converts to lowercase as is the default for pgsql when you don't have double quotes.

The solution would be write queries with quotes when using pgsql.

**Comment by noloop** [comment]
@izniburak I believe that should look like this:

```php
$query = 'SELECT ' . '"'. $this->select . '"' . ' FROM ' . '"'. $this->from. '"';
``
This for all SQL instructions.

**Comment by izniburak** [comment]
@noloop , I see. it may make sense. I will try this on all databases that provided in PDOx. If we get any result, I will try to fix it asap.

Thanks.

**Comment by noloop** [comment]
@izniburak If you need help, I can test on MySQL and Postgres I have here.

**Comment by noloop** [comment]
@izniburak Do you work in this? Because my system is dependent of the Pdox, I'm stopped. Case don't you working, I should start work this week.

I'm asking this for don't start work that you may be ending.

**Comment by izniburak** [comment]
Hi @noloop , I'm working on it but last week i did not have any time because of personal works. I will interest this in this week. If you have time, maybe you can open a PR for it.
Thanks.

**Comment by izniburak** [comment]
@noloop I think so. Btw, Can backtick char (`) solve the this problem, you think?
